### DISCUSSION TITLE
How to distribute to nix package/flakes?

### DISCUSSION THREAD

**Discussion by tfkhdyt** [tauri-apps/tauri#11802] [Q&A]
How to distribute to nix packages/flakes? I'm a beginner in Nix so I don't know how to create nix flakes for Tauri app distribution.

**Comment by bburdette** [comment]
I haven't gotten all the way there, but you can take a look at https://github.com/bburdette/tauri-zknotes.  There's a flake.nix.  That's just for development but does have necessary deps in it for desktop.  It uses a subproject zknotes that has a full nix build, but just of a web server version not the tauri one.  

If you're wanting to build an android app, its gonna be tough.  I did seemingly get that working at one time, but then it broke - probably something to do with nix-ld.  I thought it was working but then it didn't work on another machine, and then it quit working on the first machine and now I'm doing my android builds on debian lol.  More here https://nixos.wiki/wiki/Android.  If you have success with that let me know.  

Some other stuff in that flake: fenix for getting a specific rust compiler version.  naersk for building a rust project.  You can combine the two with an `override`, see the fenix readme for an example.
